mmdeploy/codebase/mmdet/deploy/object_detection_model.py

```diff
@@ -604,6 +604,71 @@ class labels of shape [N, num_det].
         return dets, labels.to(torch.int32)
 
 
+@__BACKEND_MODEL.register_module('vacc_det')
+class VACCDetModel(End2EndModel):
+
+    def __init__(self, backend: Backend, backend_files: Sequence[str],
+                 device: str, model_cfg: Union[str, Config],
+                 deploy_cfg: Union[str, Config], **kwargs):
+        assert backend == Backend.VACC, f'only supported vacc, but give \
+            {backend.value}'
+
+        super(VACCDetModel, self).__init__(backend, backend_files, device,
+                                           deploy_cfg, **kwargs)
+        # load cfg if necessary
+        model_cfg = load_config(model_cfg)[0]
+        self.model_cfg = model_cfg
+
+    def forward(self,
+                inputs: torch.Tensor,
+                data_samples: Optional[List[BaseDataElement]] = None,
+                mode: str = 'predict',
+                **kwargs) -> Any:
+        """The model forward.
+
+        Args:
+            inputs (torch.Tensor): The input tensors
+            data_samples (List[BaseDataElement], optional): The data samples.
+                Defaults to None.
+            mode (str, optional): forward mode, only support `predict`.
+        Returns:
+            Any: Model output.
+        """
+        assert mode == 'predict', 'Deploy model only allow mode=="predict".'
+        inputs = inputs.contiguous()
+        outputs = self.predict(inputs)
+        ret = self._get_bboxes(outputs[0], [i.metainfo for i in data_samples])
+        for i in range(len(ret)):
+            data_samples[i].pred_instances = ret[i]
+        return data_samples
+
+    def predict(self, imgs: Tensor, *args, **kwargs) -> List:
```

Comment:
Is `predict` a duplicate of base class `End2EndModel`?

Reply:
Yes. I rewrite this function

Reply:
But it is a duplicate of `forward_test` of `End2EndModel`. I mean, you can directly call it instead of writing a new one.

Reply:
fix~same as rknn type